day_one/part_one.go

```diff
@@ -1,84 +1,118 @@
 package main
 
 import (
+	"bufio"
 	_ "embed"
 	"fmt"
+	"io"
 	"log"
-	"sort"
+	"os"
 	"strconv"
 	"strings"
-)
-
-//go:embed input.txt
-var input string
 
-func init() {
-	// do this in init (not main) so test file has same input
-	input = strings.TrimRight(input, "\n")
-	if len(input) == 0 {
-		panic("empty input.txt file")
-	}
-}
+	"golang.org/x/exp/slices"
+)
 
 func main() {
-	parsedInput := parseInput(input)
+	parsedInput, err := ParseFile(os.Args[1])
+	if err != nil {
+		log.Fatalf("parse input: %v", err)
+	}
 
-	elves := groupElves(parsedInput)
+	elves := GroupElves(parsedInput)
 	for _, elf := range elves {
 		fmt.Println(elf)
 	}
 
-	elf := findElfWithMostest(elves)
+	elf := TopElf(elves)
 	fmt.Println("Elf with the mostest: ", elf)
 
-	topThreeElves := findThreeElvesWithTheMostest(elves)
-
+	topThreeElves := TopThreeElves(elves)
 	fmt.Println("Top three elves: ", topThreeElves)
 }
 
-func findThreeElvesWithTheMostest(elves []int) []int {
-	elves = sortElves(elves)
-	return elves[:3]
-}
+func TopThreeElves(elves []int) []int {
```

Comment:
Is there a reason you are using capitals here?

From what I was reading capitals is for exporting and other than maybe the ParseFile i wouldnt think we would care to export anything?

Reply:
I changed the test file to be in a different package. It's very common to have the test package be different from the one it's testing, but because of that the functions can't be tested if they're not exported. Not all of the functions need direct testing, though, so some could be unexported for sure. That also gives you more control over how they're called because you're the only one calling them then.

On a similar topic, it's quite common to move most logic into non-`main` packages and then make the `main` package a thin wrapper around it. That makes testing easier, but it's quite possible to test it like this, too. It's pretty easy to do because of how modules and packages work, but it's probably overkill for this project.

Reply:
That makes sense. I was using the `Generate Unit Tests for file` command in VS Code.